src/framework/accessibility/internal/accessibilitycontroller.cpp

```diff
@@ -47,20 +47,21 @@ AccessibilityController::~AccessibilityController()
     unreg(this);
 }
 
+QAccessibleInterface* AccessibilityController::accessibleInterface(QObject*)
+{
+    return static_cast<QAccessibleInterface*>(new AccessibleItemInterface(s_rootObject));
+}
+
 static QAccessibleInterface* muAccessibleFactory(const QString& classname, QObject* object)
 {
-#ifdef Q_OS_MAC
-    if (classname == QLatin1String("QQuickWindow")) {
-        return static_cast<QAccessibleInterface*>(new AccessibleItemInterface(s_rootObject));
+    auto accessibleInterfaceRegister = mu::modularity::ioc()->resolve<IAccessibleInterfaceRegister>("accessibility");
+    if (!accessibleInterfaceRegister) {
```

Comment:
This function is called very often, and `ioc()->resolve` is quite an overhead call, it is better to do it once

Reply:
fixed